### PR TITLE
[FLINK-34643][tests] Fix JobIDLoggingITCase

### PR DESCRIPTION
**Pull request by rkhachatryan**
## What is the purpose of the change

*(For example: This pull request makes task deployment go through the blob server, rather than through RPC. That way we avoid re-transferring them on each deployment (during recovery).)*

## Brief change log

*(for example:)*
  - *The TaskInfo is stored in the blob store on job creation time as a persistent artifact*
  - *Deployments RPC transmits only the blob storage reference*
  - *TaskManagers retrieve the TaskInfo from the blob cache*

## Verifying this change

Please make sure both new and modified tests in this PR follows the conventions defined in our code quality guide: https://flink.apache.org/contributing/code-style-and-quality-common.html#testing

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (100MB)*
  - *Extended integration test for recovery after master (JobManager) failure*
  - *Added test that validates that TaskInfo is transferred only once across recoveries*
  - *Manually verified the change by running a 4 node cluster with 2 JobManagers and 4 TaskManagers, a stateful streaming program, and killing one JobManager and two TaskManagers during the execution, verifying that recovery happens correctly.*

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / no)
  - The serializers: (yes / no / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / no / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (yes / no / don't know)
  - The S3 file system connector: (yes / no / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

File: flink-tests/src/test/java/org/apache/flink/test/misc/JobIDLoggingITCase.java [hunk 1]
```diff
@@ -116,18 +116,55 @@ public void testJobIDLogging(@InjectClusterClient ClusterClient<?> clusterClient
...
+        // (1e4894ed9ef9533cac2effb6f7f3e6b3_cbc357ccb763df2852fee8c4fc7d55f2_0_0), deploy into slot
+        // with allocation id 0d0aa777e9ac73047d35ec6657270462.,
+        // - Operator event for
+        // 1e4894ed9ef9533cac2effb6f7f3e6b3_cbc357ccb763df2852fee8c4fc7d55f2_0_0 -
+        // cbc357ccb763df2852fee8c4fc7d55f2,
+        // - Trigger checkpoint 1@1710170116122 for
+        // 1e4894ed9ef9533cac2effb6f7f3e6b3_cbc357ccb763df2852fee8c4fc7d55f2_0_0.,
         assertJobIDPresent(
                 jobID,
-                9,
```
[the remark this answers is not in the document]
Comment (rkhachatryan) [reply flink-tests/src/test/java/org/apache/flink/test/misc/JobIDLoggingITCase.java:121]: Wouldn't it be easier to break then?

[the remark this answers is not in the document]
Comment (rkhachatryan) [reply flink-tests/src/test/java/org/apache/flink/test/misc/JobIDLoggingITCase.java:121]: Done in 9bbb8ff.

File: flink-tests/src/test/java/org/apache/flink/test/misc/JobIDLoggingITCase.java [hunk 2]
```diff
@@ -205,6 +268,8 @@ private static JobID runJob(ClusterClient<?> clusterClient) throws Exception {
         // wait for all tasks ready and then checkpoint
         while (true) {
             try {
+                clusterClient.triggerCheckpoint(jobId, CheckpointType.DEFAULT).get();
+                // get checkpoint notification
```
Comment (XComp) [inline flink-tests/src/test/java/org/apache/flink/test/misc/JobIDLoggingITCase.java:272]: why do need a second checkpoint trigger to get the checkpoint notification? because the 2nd checkpoint is waiting for the first one to finish? :thinking:

File: flink-test-utils-parent/flink-test-utils-junit/src/main/java/org/apache/flink/testutils/logging/LoggerAuditingExtension.java [hunk 3]
```diff
@@ -81,7 +81,9 @@ public void beforeEach(ExtensionContext context) throws Exception {
                 new AbstractAppender("test-appender", null, null, false, Property.EMPTY_ARRAY) {
                     @Override
                     public void append(LogEvent event) {
-                        loggingEvents.add(event.toImmutable());
+                        if (event.toImmutable() != null) {
+                            loggingEvents.add(event.toImmutable());
+                        }
```
Comment (XComp) [inline flink-test-utils-parent/flink-test-utils-junit/src/main/java/org/apache/flink/testutils/logging/LoggerAuditingExtension.java:86]: It's not clear to me why this change is necessary. The `LogEvent` interface doesn't indicate that `null` is a valid return value. :thinking: You might want to add a comment here if we really need to check for `null` values here.

File: flink-tests/src/test/java/org/apache/flink/test/misc/JobIDLoggingITCase.java
Comment (XComp) [inline flink-tests/src/test/java/org/apache/flink/test/misc/JobIDLoggingITCase.java:1]: Just to minor nitty general comments on JUnit5 testing:
* JUnit5 doesn't require for the test class to be public, anymore. This also removes the necessity to add JavaDoc to please checkstyle.
* Same is true for the test methods

**Review by XComp — APPROVED** [review]
LGTM :+1:  Thanks for addressing my comments, @rkhachatryan . I verified that the test is not flaky anymore. I still added a few minor comments despite the approval. Decide for yourself whether they are worth being applied before merging.

File: flink-tests/src/test/java/org/apache/flink/test/misc/JobIDLoggingITCase.java (hunk 2 above)
Comment (rkhachatryan) [reply flink-tests/src/test/java/org/apache/flink/test/misc/JobIDLoggingITCase.java:272]: I wanted to check the log message when checkpoint completion notification is received. 
That notification is not guaranteed to be sent if the job is cancelled right after the checkpoint, so I added a 2nd checkpoint.

Comment (XComp) [reply flink-tests/src/test/java/org/apache/flink/test/misc/JobIDLoggingITCase.java:269]: :+1: That might be a more descriptive comment than the one in the code right now.

File: flink-test-utils-parent/flink-test-utils-junit/src/main/java/org/apache/flink/testutils/logging/LoggerAuditingExtension.java (hunk 3 above)
Comment (rkhachatryan) [reply flink-test-utils-parent/flink-test-utils-junit/src/main/java/org/apache/flink/testutils/logging/LoggerAuditingExtension.java]: I've seen NPE logged, but don't see it anymore - will remove the check.

File: flink-tests/src/test/java/org/apache/flink/test/misc/JobIDLoggingITCase.java
Comment (rkhachatryan) [reply flink-tests/src/test/java/org/apache/flink/test/misc/JobIDLoggingITCase.java:1]: Done.
